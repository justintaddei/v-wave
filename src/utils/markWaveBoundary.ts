import { triggerIsID } from './triggerIsID'

export const markWaveBoundary = (el: HTMLElement, trigger: string | boolean) => {
  el.dataset.vWaveBoundary = triggerIsID(trigger) ? trigger : 'true'
}
