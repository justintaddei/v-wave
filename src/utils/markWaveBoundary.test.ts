import { describe, expect, test } from 'vitest'
import { markWaveBoundary } from './markWaveBoundary'

describe('markWaveBoundary', () => {
  test('sets dataset to true when trigger is not an id', () => {
    const div = document.createElement('div')
    markWaveBoundary(div, 'auto')
    expect(div.dataset.vWaveBoundary).toBe('true')
  })
  test('sets dataset to the id when the trigger is an id', () => {
    const div = document.createElement('div')
    markWaveBoundary(div, 'stringId')
    expect(div.dataset.vWaveBoundary).toBe('stringId')
  })
})
