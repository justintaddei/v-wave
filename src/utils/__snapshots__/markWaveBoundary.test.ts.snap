// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`markWaveBoundary > sets dataset to the id when the trigger is an id 1`] = `
<div
  data-v-wave-boundary="stringId"
/>
`;

exports[`markWaveBoundary > sets dataset to true when trigger is not an id 1`] = `
<div
  data-v-wave-boundary="true"
/>
`;
